services/brig/test/integration/API/User/Util.hs

```diff
@@ -0,0 +1,341 @@
+setProperty brig u k v = put $ brig
+    . paths ["/properties", k]
+    . zUser u
+    . zConn "conn"
+    . contentJson
+    . body (RequestBodyLBS $ encode v)
+
+getProperty :: Brig -> UserId -> ByteString -> Http ResponseLBS
+getProperty brig u k = get $ brig
+    . paths ["/properties", k]
+    . zUser u
+
+deleteProperty :: Brig -> UserId -> ByteString -> Http ResponseLBS
+deleteProperty brig u k = delete $ brig
+    . paths ["/properties", k]
+    . zConn "conn"
+    . zUser u
+
+countCookies :: Brig -> UserId -> CookieLabel -> Http (Maybe Int)
+countCookies brig u label = do
+    r <- get ( brig
+             . path "/cookies"
+             . queryItem "labels" (toByteString' label)
+             . header "Z-User" (toByteString' u)
+             ) <!! const 200 === statusCode
+    return $ Vec.length <$> (preview (key "cookies" . _Array) =<< asValue r)
+
+assertConnections :: Brig -> UserId -> [ConnectionStatus] -> Http ()
+assertConnections brig u cs = listConnections brig u !!! do
+    const 200 === statusCode
+    const (Just True) === fmap (check . map status . clConnections) . decodeBody
+  where
+    check xs = all (`elem` xs) cs
+    status c = ConnectionStatus (ucFrom c) (ucTo c) (ucStatus c)
+
+assertEmailVisibility :: Brig -> User -> User -> Bool -> Http ()
+assertEmailVisibility brig a b visible =
+    get (brig . paths ["users", pack . show $ userId b] . zUser (userId a)) !!! do
+        const 200 === statusCode
+        if visible
+            then const (Just (userEmail b)) === fmap userEmail . decodeBody
+            else const Nothing === (userEmail <=< decodeBody)
+
+uploadAddressBook :: Brig -> UserId -> AddressBook -> MatchingResult -> Http ()
+uploadAddressBook b u a m =
+    post ( b
+         . path "/onboarding/v3"
+         . contentJson
+         . zUser u
+         . body (RequestBodyLBS $ encode a)
+         ) !!! do
+            const 200 === statusCode
+            const (Just (f m)) === (fmap f . decodeBody)
+  where
+    f :: MatchingResult -> MatchingResult
+    f (MatchingResult x y) = MatchingResult (sort x) (sort y)
+
+-- Builds expectations on the matched users/cards
+toMatchingResult :: [(UserId, Text)] -> MatchingResult
+toMatchingResult xs = MatchingResult
+                      (map (\(u, c) -> Match u (Just (CardId c)) [CardId c]) xs)
+                      (Set.toList $ Set.fromList (map fst xs))
+
+-- Hashes each entry and builds an appropriate address book
+toAddressBook :: [(Text, [Text])] -> IO AddressBook
+toAddressBook xs = do
+    Just sha <- liftIO $ getDigestByName "SHA256"
+    return . AddressBook $ fmap (toCard sha) xs
+  where
+    toCard sha (cardId, entries) = Card (Just $ CardId cardId)
+                                        (map (Entry . digestBS sha . T.encodeUtf8) entries)
+
+setHandleAndDeleteUser :: Brig -> Cannon -> User -> [UserId] -> (UserId -> HttpT IO ()) -> Http ()
+setHandleAndDeleteUser brig cannon u others execDelete = do
```

Comment:
This long helper function would perhaps be better placed in the only file it's used, i.e. `integration/API/User/Account.hs`